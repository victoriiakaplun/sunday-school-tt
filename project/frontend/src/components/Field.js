import React from 'react';

function Field({ children, type, placeholder }) {
  return (
    <div className="field">
      <label htmlFor={children} className="label">
        {children}
      </label>
      <div className="control">
        <input className="input" type={type} placeholder={placeholder} name={children} required />
      </div>
    </div>
  );
}

export default Field;
