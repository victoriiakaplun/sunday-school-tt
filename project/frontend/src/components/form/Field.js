import React from 'react';

function Field({ children, type, name, placeholder, value, onChange }) {
  return (
    <div className="field">
      <label htmlFor={children} className="label">
        {children}
      </label>
      <div className="control">
        <input
          className="input"
          type={type}
          placeholder={placeholder}
          name={name}
          onChange={onChange}
          required
        />
      </div>
    </div>
  );
}

export default Field;
