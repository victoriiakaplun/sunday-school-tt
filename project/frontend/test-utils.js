import React from 'react'; // eslint-disable-line no-unused-vars

import { initFontAwesomeLibrary } from './src/utils';

initFontAwesomeLibrary();

// import '@testing-library/jest-dom/extend-expect';

// re-export everything
export * from '@testing-library/react';
