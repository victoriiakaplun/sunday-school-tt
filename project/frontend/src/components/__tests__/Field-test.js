import { render, fireEvent, screen } from 'test-utils';
import Field from '../form/Field';

describe('Field', () => {
  it('renders correctly', () => {
    const { container } = render(<Field>entry</Field>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('handles onChange event', async () => {
    const onChangeMock = jest.fn();
    const testChange = 'test';
    render(<Field onChange={onChangeMock} />);
    const input = await screen.getByTestId('field_input');
    fireEvent.change(input, { target: { value: testChange } });

    expect(onChangeMock).toBeCalledTimes(1);
  });
});
