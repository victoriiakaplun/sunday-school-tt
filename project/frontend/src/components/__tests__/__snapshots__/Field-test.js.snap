// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders correctly: 
      <div 
      class="field">
      entry
      </div>
     1`] = `
<div
  class="field"
>
  <label
    class="label"
    for="entry"
  >
    entry
  </label>
  <div
    class="control"
  >
    <input
      class="input"
      required=""
      value=""
    />
  </div>
</div>
`;
